metrics/metrics.go

```diff
@@ -0,0 +1,35 @@
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/metrics/prometheus"
+)
+
+// MetricsServer runs and controls a HTTP pprof interface.
+type MetricsServer struct {
+	server *http.Server
+}
+
+func NewMetricsServer(port int, r metrics.Registry) *MetricsServer {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", prometheus.Handler(r))
+	p := MetricsServer{
+		server: &http.Server{
+			Addr:    fmt.Sprintf(":%d", port),
+			Handler: mux,
+		},
+	}
+	return &p
+}
+
+// Go starts the HTTP pprof in the background.
+func (p *MetricsServer) Go() {
```

Comment:
```suggestion
func (p *MetricsServer) Listen() {
    log.Info("metrics server stopped", "err", p.server.ListenAndServe())
}
```

Also, in this case, it's totally fine for the caller to take the responsibility to call it in a goroutine rather than do it here.

So it would be executed like this:
```go
go metrics.NewMetricsServer(metricsPort, gethmetrics.DefaultRegistry).Listen()
```

Reply:
Sure, why not.